### PR TITLE
Fixing snapshots 

### PR DESCRIPTION
**Pull request by mrickard**
Using a subset of readablespan props for snapshots/tracking, and adding service name in constructor for naming the resource

### CODE REVIEW

File: azureLogIngestion/index.ts [hunk 1]
```diff
@@ -1,21 +1,24 @@
...
-const otelAdapter = new OpenTelemetryAdapter(apiKey)
+let otelAdapter: OpenTelemetryAdapter
 
 const eventHubTrigger: AzureFunction = async function (context: Context, eventHubMessages: any): Promise<void> {
     if (debug) {
         context.log(`Eventhub trigger function called for message eventHubMessages ${eventHubMessages}`)
         context.log(`eventHubMessages type: ${typeof eventHubMessages}`)
     }
     if (otel) {
+        const serviceName = _.get(eventHubMessages, "records.0.resourceId", null)
```
Comment (kolanos) [inline azureLogIngestion/index.ts:19]: We have `./utils/resource` to parse this. We probably want `resourceName` only here.

File: azureLogIngestion/opentelemetry/index.ts [hunk 2]
```diff
@@ -162,7 +222,7 @@ export default class OpenTelemetryAdapter {
         span.end(endTimeHrFromDuration(appSpan.timestamp, appSpan.durationMs))
         // OT batch processor doesn't give access to current batch size
         // or batch content. This lets us do snapshot tests.
-        this.currentBatch.push(span)
+        this.currentBatch.push(loggableSpan(span))
```
Comment (kolanos) [inline azureLogIngestion/opentelemetry/index.ts:225]: Is there a way to detect we're in a test suite and only do this there? Not a big deal for now.

**Review by kolanos — CHANGES_REQUESTED** [review]
Only minor things.

File: azureLogIngestion/opentelemetry/index.ts (hunk 2 above)
Comment (mrickard) [reply azureLogIngestion/opentelemetry/index.ts:225]: @kolanos Arguably we could set an env var in one of the tests and base this off of that. 

I do like the idea of pushing something into the currentBatch, though, as a way of detecting whether or not there are otel spans to flush.

File: azureLogIngestion/index.ts [hunk 3]
```diff
@@ -1,21 +1,25 @@
...
-const otelAdapter = new OpenTelemetryAdapter(apiKey)
+let otelAdapter: OpenTelemetryAdapter
 
 const eventHubTrigger: AzureFunction = async function (context: Context, eventHubMessages: any): Promise<void> {
     if (debug) {
         context.log(`Eventhub trigger function called for message eventHubMessages ${eventHubMessages}`)
         context.log(`eventHubMessages type: ${typeof eventHubMessages}`)
     }
     if (otel) {
+        const { resourceId } = eventHubMessages.records[0]
```
Comment (kolanos) [inline azureLogIngestion/index.ts]: First, I wouldn't do it this way. This function is going to get records for all sorts of resources. So this won't be reliable.

Second, why are we parsing this value and not `resourceId`? That's what `parse()` is expecting...

**Review by kolanos — CHANGES_REQUESTED** [review]

File: azureLogIngestion/index.ts (hunk 3 above)
Comment (mrickard) [reply azureLogIngestion/index.ts]: resourceId isn't available to us yet--if we're going to get it here at all, it's going to be buried in eventHubMessages. So yeah, this isn't optimal. I do agree that we may have to extend traceProvider to let us set this later.

**Review by kolanos — APPROVED** [review]

**Pull request merged**